database/src/main/dataset_schema/add_schema.sql

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+DROP FUNCTION IF EXISTS dataset_schema.add_schema(TEXT, INTEGER, TEXT, JSONB, TEXT);
+
+CREATE OR REPLACE FUNCTION dataset_schema.add_schema(
+    IN i_schema_name        TEXT,
+    IN i_schema_version     INTEGER,
+    IN i_schema_description TEXT,
+    IN i_fields             JSONB,
+    IN i_user_name          TEXT,
+    OUT status              INTEGER,
+    OUT status_text         TEXT,
+    OUT id_schema           BIGINT
+) RETURNS record AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: jobs_configuration.add_schema(5)
+--      Stores a new version of the schema.
+--      The i_schema_version has to be an increment of the latest version of an existing schema or 1
+--
+-- Parameters:
+--      i_schema_name               - name of the schema
+--      i_schema_version            - version of the schema
+--      i_schema_description        - description of the schema
+--      i_fields                    - the fields the schema consist of
+--      i_user_name                 - the user who submitted the changes
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_schema           - id of the newly created schema record
+--
+-- Status codes:
+--      201     - OK
+--      403     - Schema already exists
+--      404     - Schema version wrong
+--      405     - Schema has been disabled
+--      406     - Schema is locked
```

Comment:
This looks like HTTP/REST status codes compatible use, but their usage is quite non-standard. We should not be inventing status codes, I think. How about:
```sql
--      201     - OK
--      409     - (conflict) Schema already exists
--      404     - (not found) Schema version wrong
--      403     - (forbidden) Schema has been disabled
--      403     - (forbidden) Schema is locked
``` 
By that I mean:
 - 404 is clear - incorrect path = not found
 - 409 Conflict is typical for out-of-sync edits
 - for the rest, 403 - forbidden (i.e. request is valid, but the state of the entity does not allow to continue. Here, the difference would have to be documented by the `status_text`.

If the above is unsatisfactory, perhaps `432 Locked` from WebDAV use could be misused, it is pretty close:

```sql
--      409     - (conflict) Schema already exists
--      404     - (not found) Schema version wrong
--      403     - (forbidden) Schema has been disabled
--      423     - (locked) Schema is locked (maybe)
``` 
A relevant info source: https://www.restapitutorial.com/httpstatuscodes.html